### ISSUE TITLE
DEB packages and Docker image do not use --locked like public install instructions direct

### ISSUE THREAD

**Issue by ximon18** [NLnetLabs/krill#467]
The published installation instructions say to use `cargo install --locked` and the `Dockerile` also uses `cargo build --locked`. However, the DEB package built by `cargo deb` apparently doesn't use `--locked` as [its docs say](https://github.com/mmstick/cargo-deb#custom-build-flags):

> Flags after -- are passed to cargo build, so you can use options such as -Z, --frozen, and --locked. Please use that only for features that cargo-deb doesn't support natively.

And our `pky.yml` GitHub Actions workflow invokes cargo deb like so:

```
        DEB_VER="${DEB_KRILL_VER}-1${OS_REL}"
        cargo deb --variant ${OS_NAME}-${OS_REL} --deb-version ${DEB_VER} -v
```

**Comment by ximon18** [comment]
@timbru: Should I submit a PR to pass --locked to cargo deb ? Without it the released DEB could end up using newer dependency versions than will be installed by users who use cargo install --locked as we advise in our installation instructions. 

Note: I see also now that the `Dockerfile` for Krill also doesn't have `--locked`, while for Routinator it does.

**Comment by timbru** [comment]
I think we should use `--locked` everywhere to assure that we get the same builds and behaviour. And... we should monitor for security updates in libraries we depend on and do minor release when needed.

**Comment by ximon18** [comment]
I will submit a PR for the former for the DEB and Docker packaging.

For the latter, we could add a GitHub Actions workflow that runs `cargo audit`.

**Comment by ximon18** [comment]
FYI the Oso project runs `cargo audit` nightly or on change to `Cargo.lock` with GH Actions [like this](https://github.com/osohq/oso/blob/main/.github/workflows/audit.yml).

For the Krill `v0.8.1-bis` branch Cargo audit reports 6 vulnerabilities and 9 warnings:

```
$ cargo audit
    Fetching advisory database from `https://github.com/RustSec/advisory-db.git`
      Loaded 280 security advisories (from /home/ximon/.cargo/advisory-db)
    Updating crates.io index
    Scanning Cargo.lock for vulnerabilities (320 crate dependencies)
Crate:         generator
Version:       0.6.23
Title:         Generators can cause data races if non-Send types are used in their generator functions
Date:          2020-11-16
ID:            RUSTSEC-2020-0151
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0151
Solution:      Upgrade to >=0.7.0
Dependency tree: 
generator 0.6.23
└── loom 0.3.6
    └── sharded-slab 0.1.0
        └── tracing-subscriber 0.2.15
            └── oso 0.11.2
                └── krill 0.8.2-bis

Crate:         generic-array
Version:       0.12.3
Title:         arr! macro erases lifetimes
Date:          2020-04-09
ID:            RUSTSEC-2020-0146
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0146
Solution:      Upgrade to >=0.8.4, <0.9.0 OR >=0.9.1, <0.10.0 OR >=0.10.1, <0.11.0 OR >=0.11.2, <0.12.0 OR >=0.12.4, <0.13.0 OR >=0.13.3
Dependency tree: 
generic-array 0.12.3

Crate:         hyper
Version:       0.13.8
Title:         Multiple Transfer-Encoding headers misinterprets request payload
Date:          2021-02-05
ID:            RUSTSEC-2021-0020
URL:           https://rustsec.org/advisories/RUSTSEC-2021-0020
Solution:      Upgrade to >=0.14.3 OR >=0.13.10, <0.14.0 OR >=0.12.36, <0.13.0
Dependency tree: 
hyper 0.13.8

Crate:         ordered-float
Version:       1.1.0
Title:         ordered_float:NotNan may contain NaN after panic in assignment operators
Date:          2020-12-06
ID:            RUSTSEC-2020-0082
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0082
Solution:      Upgrade to >=1.1.1, <2.0.0 OR >=2.0.1
Dependency tree: 
ordered-float 1.1.0
└── serde-value 0.6.0
    └── openidconnect 2.0.0-beta.1
        └── krill 0.8.2-bis

Crate:         smallvec
Version:       0.6.13
Title:         Buffer overflow in SmallVec::insert_many
Date:          2021-01-08
ID:            RUSTSEC-2021-0003
URL:           https://rustsec.org/advisories/RUSTSEC-2021-0003
Solution:      Upgrade to >=0.6.14, <1.0.0 OR >=1.6.1
Dependency tree: 
smallvec 0.6.13

Crate:         smallvec
Version:       1.4.2
Title:         Buffer overflow in SmallVec::insert_many
Date:          2021-01-08
ID:            RUSTSEC-2021-0003
URL:           https://rustsec.org/advisories/RUSTSEC-2021-0003
Solution:      Upgrade to >=0.6.14, <1.0.0 OR >=1.6.1
Dependency tree: 
smallvec 1.4.2

Crate:         dirs
Version:       1.0.5
Warning:       unmaintained
Title:         dirs is unmaintained, use dirs-next instead
Date:          2020-10-16
ID:            RUSTSEC-2020-0053
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0053
Dependency tree: 
dirs 1.0.5
└── term 0.5.2
    ├── lalrpop 0.19.1
    │   ├── polar-core 0.11.2
    │   │   └── oso 0.11.2
    │   │       └── krill 0.8.2-bis
    │   └── basic-cookies 0.1.4
    │       └── krill 0.8.2-bis
    └── ascii-canvas 2.0.0
        └── lalrpop 0.19.1

Crate:         failure
Version:       0.1.8
Warning:       unmaintained
Title:         failure is officially deprecated/unmaintained
Date:          2020-05-02
ID:            RUSTSEC-2020-0036
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0036
Dependency tree: 
failure 0.1.8
└── cookie_store 0.7.0
    └── reqwest 0.9.24

Crate:         net2
Version:       0.2.35
Warning:       unmaintained
Title:         `net2` crate has been deprecated; use `socket2` instead
Date:          2020-05-01
ID:            RUSTSEC-2020-0016
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0016
Dependency tree: 
net2 0.2.35
├── miow 0.2.1
│   └── mio 0.6.22
│       ├── tokio-tcp 0.1.4
│       │   ├── tokio 0.1.22
│       │   └── hyper 0.12.36
│       ├── tokio-reactor 0.1.12
│       │   ├── tokio-tcp 0.1.4
│       │   ├── tokio 0.1.22
│       │   └── hyper 0.12.36
│       ├── tokio 0.2.22
│       └── tokio 0.1.22
├── mio 0.6.22
└── hyper 0.12.36

Crate:         term
Version:       0.5.2
Warning:       unmaintained
Title:         term is looking for a new maintainer
Date:          2018-11-19
ID:            RUSTSEC-2018-0015
URL:           https://rustsec.org/advisories/RUSTSEC-2018-0015
Dependency tree: 
term 0.5.2
├── lalrpop 0.19.1
│   ├── polar-core 0.11.2
│   │   └── oso 0.11.2
│   │       └── krill 0.8.2-bis
│   └── basic-cookies 0.1.4
│       └── krill 0.8.2-bis
└── ascii-canvas 2.0.0
    └── lalrpop 0.19.1

Crate:         miow
Version:       0.2.1
Warning:       yanked
Dependency tree: 
miow 0.2.1
└── mio 0.6.22
    ├── tokio-tcp 0.1.4
    │   ├── tokio 0.1.22
    │   └── hyper 0.12.36
    ├── tokio-reactor 0.1.12
    │   ├── tokio-tcp 0.1.4
    │   ├── tokio 0.1.22
    │   └── hyper 0.12.36
    ├── tokio 0.2.22
    └── tokio 0.1.22

Crate:         net2
Version:       0.2.35
Warning:       yanked

Crate:         pin-project-lite
Version:       0.1.11
Warning:       yanked
Dependency tree: 
pin-project-lite 0.1.11

Crate:         pin-project-lite
Version:       0.2.4
Warning:       yanked
Dependency tree: 
pin-project-lite 0.2.4

Crate:         socket2
Version:       0.3.15
Warning:       yanked
Dependency tree: 
socket2 0.3.15
└── hyper 0.13.8

error: 6 vulnerabilities found!
warning: 9 allowed warnings found
```

**Comment by ximon18** [comment]
As the title issue was fixed by PR #468 and the separate `cargo audit` proposal raised here is also tracked in #366 I'm going to close this issue.
